roles/openshift_logging/README.md

```diff
@@ -222,3 +222,27 @@ The corresponding openshift\_logging\_mux\_* parameters are below.
 - `openshift_logging_mux_remote_syslog_tag_key`: If string specified, use this field from the record to set the key field on the syslog message
 - `openshift_logging_mux_remote_syslog_use_record`: Set `true` to use the severity and facility from the record, defaults to `false`
 - `openshift_logging_mux_remote_syslog_payload_key`: If string is specified, use this field from the record as the payload on the syslog message
+
+Image update procedure
+----------------------
+
+An upgrade of the logging stack from older version to newer should always be performed by calling appropriate ansible playbook and setting required ansible variables in your inventory as documented in https://docs.openshift.org/.
+
+There could be a situation where user may want to update only the logging images and not the entire logging stack setup.
```

Comment:
This isn't what I had in mind... let me write up an example.

Reply:
To determine the current version of images being used you can do the following:
$ oc describe pod/logging-fluentd-fh458
. . .
Containers:
  fluentd-elasticsearch:
    . . .
    Image ID:		docker-pullable://brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/openshift3/logging-fluentd@sha256:4346f0aa9694f32735115705ad324803b1a6ff08343c3288f7a62c3a5cb70495
    . . .

$ docker images
REPOSITORY                                                                        TAG                 IMAGE ID            CREATED             SIZE
brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/openshift3/logging-fluentd   v3.7                ff2e249fc45a        About an hour ago   235.2 MB

$ docker inspect ff2e249fc45a
[
    {
        . . .
        "RepoDigests": [
            "brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/openshift3/logging-fluentd@sha256:4346f0aa9694f32735115705ad324803b1a6ff08343c3288f7a62c3a5cb70495"
        ],
        . . .
        "Config": {
            . . .
            "Labels": {
                . . .
                "build-date": "2017-10-12T14:38:22.414827",
                . . . 
                "release": "0.143.3.0",
                . . .
                "url": "https://access.redhat.com/containers/#/registry.access.redhat.com/openshift3/logging-fluentd/images/v3.7.0-0.143.3.0",
                . . .
                "version": "v3.7.0"
            }
        },
        . . .

 
 If necessary to pull a new image:
 $ docker pull brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/openshift3/logging-fluentd:v3.7
 
 It is recommended that you now rerun the openshift_logging playbook to ensure that any necessary config changes are also picked up.
 
 To manually redeploy your pod you can do the following (not recommended):
   For a DC you can do:
     oc rollout latest <dc_name>
     
   For a RC you can scale down and scale back up
   For a DS you can delete the pod or unlabel and relabel your node

Reply:
There are two things I dislike about your suggestion:
1) if user wants to update an image:tag they already have pulled, they **need** to login to every single machine and do a manual `docker pull IMAGE`

2) when they run the playbook afterward, if there were no changes to dc/ds/rs, they need to do the `oc rollout` anyways because the task of redeploying dc/ds/rs gets skipped

Please, reconsider the temporary change of `imagePullPolicy: Always` and then revert back, I think it has a lot fewer opportunities to break than what you are proposing. Or at least let me know what specifically you find wrong with it.

Reply:
True, my concern with updating to use `imagePullPolicy: Always` doing a deployment and then reverting and redeploying is that is a lot of extra work that Elasticsearch has to do with starting and stopping. That was why I was thinking we would redeploy once and not update the imagePullPolicy. The `docker pull IMAGE` could easily be scripted for users that had a large number of nodes.

Interestingly, we moved away from ImageStreams because we had issues where customers did not have the most up to date images... but I agree, it would be more simple.

Reply:
@ewolinetz, see issue https://github.com/openshift/openshift-ansible/issues/5755, this would allow us to use proper handlers in ansible to avoid unwanted restarts, properly restarted services as each service requires.

Reply:
@portante true, however I think that is independent to my concern with patching to change the image pull policy twice for each deployment object. We would need the pod deployers to actually trigger to resolve the issue of out of date images being pulled...

Reply:
@ewolinetz, yes, I think your concern is warranted. I did not review this code closely at the time of the above comment, so please don't take my comment in support of this particular change.  My comment was made to consider the fact that with triggers enabled, image changes are not the only reason for pods to restart, since `IfNotPresent` is only about when an image pull is made.

Reply:
@ewolinetz ack, I will update the README as you suggested and start preparing the ImageStreams for the next release